### PR TITLE
RPC-AIO Pipeline Connect Slave

### PR DESCRIPTION
**Pull request by hughsaunders**
Add reusable step for setting up an existing node as a jenkins slave.

Depends on #4

Connects rcbops/u-suk-dev#1066

### CODE REVIEW

File: playbooks/remove-jenkins-slave.yml [hunk 1]
```diff
@@ -0,0 +1,13 @@
+- hosts: job_nodes
```
Comment (mattt416) [inline playbooks/remove-jenkins-slave.yml]: Would this not need to run on the CIT jenkins slave itself?

Comment (hughsaunders) [inline playbooks/remove-jenkins-slave.yml]: It does, its delegated.

Comment (mattt416) [inline playbooks/remove-jenkins-slave.yml]: :eagle:

File: rpc-jobs/RPC-AIO.yml [hunk 2]
```diff
@@ -19,12 +19,21 @@
...
+          name: INSTANCE_NAME
+          default: AUTO
+          description: |
+            Set instance name or "AUTO" to have it generated based on job
+            name and build number. Beware the 63 char container name limit
+            as container names includes the host name.
+      - string:
           name: STAGES
-          default: "Allocate Resources, Cleanup"
+          default: "Allocate Resources, Connect Slave, Pause, Cleanup"
```
Comment (melissa-kam) [inline rpc-jobs/RPC-AIO.yml]: If we pause by default, will that turn into a manual burden of having to input continue for all the builds?

Comment (hughsaunders) [inline rpc-jobs/RPC-AIO.yml]: Yes! that was only there for testing, I'll remove it.

File: pipeline-steps/pub_cloud_slave_highlevel.groovy [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+      if (env.INSTANCE_NAME == "AUTO"){
+        job_name_acronym=""
+        job_words=env.JOB_NAME.split("[-_ ]")
+        for (i=0; i<job_words.size(); i++){
+          job_name_acronym += job_words[i][0]
+        }
+        build_name = "${job_name_acronym}-${env.BUILD_NUMBER}"
+      }
+      else {
+        build_name = env.JOB_NAME
```
Comment (melissa-kam) [inline pipeline-steps/pub_cloud_slave_highlevel.groovy]: This should be `env.INSTANCE_NAME`.

File: rpc-jobs/RPC-AIO.yml (hunk 2 above)
Comment (hughsaunders) [inline rpc-jobs/RPC-AIO.yml]: Done.

File: pipeline-steps/pub_cloud_slave_highlevel.groovy (hunk 3 above)
Comment (hughsaunders) [inline pipeline-steps/pub_cloud_slave_highlevel.groovy]: Good catch, fixed.

**Comment by hughsaunders** [comment]
This is now ready for review again, after a non-functional period yesterday. That was caused by me forward porting the single use slave template commit into this branch.

**Pull request merged**